aws/resource_aws_flow_log.go

```diff
@@ -18,45 +19,71 @@ func resourceAwsFlowLog() *schema.Resource {
 		Importer: &schema.ResourceImporter{
 			State: schema.ImportStatePassthrough,
 		},
+		MigrateState:  resourceAwsFlowLogMigrateState,
+		SchemaVersion: 1,
 
 		Schema: map[string]*schema.Schema{
+
 			"iam_role_arn": &schema.Schema{
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
 				ForceNew: true,
 			},
 
-			"log_group_name": &schema.Schema{
+			"log_destination": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+
+			"log_destination_type": &schema.Schema{
 				Type:     schema.TypeString,
-				Required: true,
+				Default:  ec2.LogDestinationTypeCloudWatchLogs,
+				Optional: true,
 				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					ec2.LogDestinationTypeCloudWatchLogs,
+					ec2.LogDestinationTypeS3,
+				}, false),
 			},
 
-			"vpc_id": &schema.Schema{
```

Comment:
Removing existing attributes will require a deprecation warning and support of the attributes until deprecation.